src/integrators/integrator_utils.jl

```diff
@@ -296,6 +296,187 @@ function step_reject_controller!(integrator,alg::PredictiveControllerAlgs)
   end
 end
 
+
+@inline function stepsize_controller!(integrator,alg::ExtrapolationMidpointDeuflhard)
+  # Dummy function
+  # ExtrapolationMidpointDeuflhard's stepsize scaling is stored in the cache;
+  # it is computed by  stepsize_controller_internal! (in perfom_step!) resp. stepsize_predictor!
+  # (in step_accept_controller! and step_reject_controller!)
+  zero(typeof(integrator.opts.qmax))
+end
+
+function stepsize_controller_internal!(integrator,alg::ExtrapolationMidpointDeuflhard)
+  # Standard stepsize controller
+  # Compute and save the stepsize scaling based on the latest error estimate of the current order
+  if iszero(integrator.EEst)
+    q = inv(integrator.opts.qmax)
+  else
+    # Update gamma and beta1
+    integrator.opts.beta1 = typeof(integrator.opts.beta1)(1 // (2integrator.cache.n_curr + 1))
+    integrator.opts.gamma = typeof(integrator.opts.gamma)(1 // 4)^integrator.opts.beta1
+    # Compute new stepsize scaling
+    qtmp = integrator.EEst^integrator.opts.beta1 / integrator.opts.gamma
+    @fastmath q = max(inv(integrator.opts.qmax), min(inv(integrator.opts.qmin), qtmp))
+  end
+  integrator.cache.Q[integrator.cache.n_curr - alg.n_min + 1] = q
+end
+
+function stepsize_predictor!(integrator,alg::ExtrapolationMidpointDeuflhard,n_new::Int64)
+  # Compute and save the stepsize scaling for order n_new based on the latest error estimate of the current order.
+  if iszero(integrator.EEst)
+    q = inv(integrator.opts.qmax)
+  else
+    # Initialize
+    @unpack t,EEst = integrator
+    tol = integrator.opts.internalnorm(integrator.opts.reltol,t) # Deuflhard's approach relies on EEstD ≈ ||relTol||
```

Comment:
This might have weird interactions with element-wise tolerances, but does work.

Reply:
That is a problem I have not figured out yet. The thing is that in Deuflhard's approach some sort of notion of a relative tolerance (< 1 !)  is needed for the convergence monitor and stepsize prediction.

I guess this forces the user in the case of vector valued rel. and abs. tolerance to think of these vectors as scaling vectors multiplied by `tol`. 

So far I haven't played with element-wise tolerances and don't know what the effects are...